### PR TITLE
Feature open with

### PR DESCRIPTION
**Pull request by a-szotyori**
Created methods for client apps to create an Intent for the Screen API or a Document for the Component API from an "open with" Intent.

### How to test
Run the example apps and send images and pdfs from emails, photos, downloads, etc. to the example apps.

### Merging
Automatic.

### Ticket 
Issue #88

### CODE REVIEW

File: componentapiexample/src/main/java/net/gini/android/vision/component/GiniVisionActivity.java [hunk 1]
```diff
@@ -163,14 +166,64 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            } else {
+                showFragment(getAnalysisFragment(document), R.string.title_review);
+            }
+        } catch (ImportedFileValidationException e) {
+            e.printStackTrace();
+            String message = "File cannot be analyzed";
+            if (e.getValidationError() != null) {
+                switch (e.getValidationError()) {
+                    case TYPE_NOT_SUPPORTED:
+                        message = "File type not supported.";
```
Comment (kikettas) [inline componentapiexample/src/main/java/net/gini/android/vision/component/GiniVisionActivity.java]: This messages should be extracted to a `strings` file and written in german, right?

File: componentapiexample/src/main/java/net/gini/android/vision/component/GiniVisionAppCompatActivity.java [hunk 2]
```diff
@@ -1,11 +1,15 @@
 package net.gini.android.vision.component;
```
Comment (kikettas) [inline componentapiexample/src/main/java/net/gini/android/vision/component/GiniVisionAppCompatActivity.java]: There is a lot of duplicated code in this file if you compare it with `GiniVisionActivity`. Is there a way to join them  and get rid of the duplicated code?

File: ginivision/src/main/java/net/gini/android/vision/GiniVision.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        }
+        return giniVisionIntent;
+    }
+
+    @NonNull
+    public static Document createDocumentForImportedFile(@NonNull final Intent intent,
+            @NonNull final Context context) throws ImportedFileValidationException {
+        final Uri uri = IntentHelper.getUri(intent);
+        if (uri == null) {
+            throw new ImportedFileValidationException("todo description no uri");
```
Comment (kikettas) [inline ginivision/src/main/java/net/gini/android/vision/GiniVision.java]: I saw that `ImportedFileValidationException` stack trace is printed in other classes, so maybe we should add a proper description here.

File: ginivision/src/main/java/net/gini/android/vision/analysis/AnalysisActivity.java [hunk 4]
```diff
@@ -164,6 +164,10 @@ public void noExtractionsFound() {
             startActivity(noResultsActivity);
             setResult(RESULT_NO_EXTRACTIONS);
             finish();
```
Comment (kikettas) [inline ginivision/src/main/java/net/gini/android/vision/analysis/AnalysisActivity.java]: This `finish()` call is duplicated and could be moved to the end of the method.

File: componentapiexample/src/main/java/net/gini/android/vision/component/GiniVisionAppCompatActivity.java (hunk 2 above)
Comment (a-szotyori) [reply componentapiexample/src/main/java/net/gini/android/vision/component/GiniVisionAppCompatActivity.java]: Clients will have to study either one or the other depending on whether they use the support library or not. I would not abstract the code away for the example. I would like to keep everything in one file for each case. They share logic but shouldn't share code. Presenting an independent example for each case is more important than maintainability.

File: ginivision/src/main/java/net/gini/android/vision/GiniVision.java (hunk 3 above)
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/GiniVision.java]: Whew, thanks for noticing this!

**Pull request merged**